ni_measurementlink_service/measurement/info.py

```diff
@@ -67,26 +67,107 @@ class TypeSpecialization(enum.Enum):
 class DataType(enum.Enum):
     """Enum that represents the supported data types."""
 
-    Int32 = (type_pb2.Field.TYPE_INT32, False, TypeSpecialization.NoType)
-    Int64 = (type_pb2.Field.TYPE_INT64, False, TypeSpecialization.NoType)
-    UInt32 = (type_pb2.Field.TYPE_UINT32, False, TypeSpecialization.NoType)
-    UInt64 = (type_pb2.Field.TYPE_UINT64, False, TypeSpecialization.NoType)
-    Float = (type_pb2.Field.TYPE_FLOAT, False, TypeSpecialization.NoType)
-    Double = (type_pb2.Field.TYPE_DOUBLE, False, TypeSpecialization.NoType)
-    Boolean = (type_pb2.Field.TYPE_BOOL, False, TypeSpecialization.NoType)
-    String = (type_pb2.Field.TYPE_STRING, False, TypeSpecialization.NoType)
-    Pin = (type_pb2.Field.TYPE_STRING, False, TypeSpecialization.Pin)
-    Path = (type_pb2.Field.TYPE_STRING, False, TypeSpecialization.Path)
-    Enum = (type_pb2.Field.TYPE_ENUM, False, TypeSpecialization.Enum)
-
-    Int32Array1D = (type_pb2.Field.TYPE_INT32, True, TypeSpecialization.NoType)
-    Int64Array1D = (type_pb2.Field.TYPE_INT64, True, TypeSpecialization.NoType)
-    UInt32Array1D = (type_pb2.Field.TYPE_UINT32, True, TypeSpecialization.NoType)
-    UInt64Array1D = (type_pb2.Field.TYPE_UINT64, True, TypeSpecialization.NoType)
-    FloatArray1D = (type_pb2.Field.TYPE_FLOAT, True, TypeSpecialization.NoType)
-    DoubleArray1D = (type_pb2.Field.TYPE_DOUBLE, True, TypeSpecialization.NoType)
-    BooleanArray1D = (type_pb2.Field.TYPE_BOOL, True, TypeSpecialization.NoType)
-    StringArray1D = (type_pb2.Field.TYPE_STRING, True, TypeSpecialization.NoType)
-    PinArray1D = (type_pb2.Field.TYPE_STRING, True, TypeSpecialization.Pin)
-    PathArray1D = (type_pb2.Field.TYPE_STRING, True, TypeSpecialization.Path)
-    EnumArray1D = (type_pb2.Field.TYPE_ENUM, True, TypeSpecialization.Enum)
+    Int32 = 0
+    Int64 = 1
+    UInt32 = 2
+    UInt64 = 3
+    Float = 4
+    Double = 5
+    Boolean = 6
+    String = 7
+    Pin = 8
+    Path = 9
+    Enum = 10
+
+    Int32Array1D = 100
+    Int64Array1D = 101
+    UInt32Array1D = 102
+    UInt64Array1D = 103
+    FloatArray1D = 104
+    DoubleArray1D = 105
+    BooleanArray1D = 106
+    StringArray1D = 107
+    PinArray1D = 108
+    PathArray1D = 109
+    EnumArray1D = 110
+
+
+class DataTypeInfo(NamedTuple):
+    """Class that represents the information for each of the DataType enum values.
+
+    Attributes
+    ----------
+        grpc_field_type (type_pb2.Field.Kind): Field.Kind associated with the DataType
+
+        repeated (bool): Whether the DataType is a repeated field
+
+        type_specialization (TypeSpecialization): Specific type when value_type
+        can have more than one use
+
+    """
+
+    grpc_field_type: type_pb2.Field.Kind.ValueType
+    repeated: bool
+    type_specialization: TypeSpecialization = TypeSpecialization.NoType
+
+
+Int32TypeInfo = DataTypeInfo(type_pb2.Field.TYPE_INT32, False)
```

Comment:
These are constants, not types, so they should be in UPPER_SNAKE_CASE

https://pep8.org/#constants

Reply:
This also applies to the enum values but it's too late to fix those

Reply:
Also, these constants aren't accessed from outside this module, so you can make them private. Or you can delete them and move the data into the dictionary initializer.

Reply:
I'm going to delete these and use the dictionary initializer, but how do you make a constant private in Python?

Reply:
Just put them in a private module?

Reply:
Use a leading underscore.
https://pep8.org/#public-and-internal-interfaces